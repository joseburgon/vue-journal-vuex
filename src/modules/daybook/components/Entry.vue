<template>
  <div @click="$router.push({ name: 'entry', params: { id: 10 } })" class="entry-container mb-3 pointer p-2">
    <div class="entry-title d-flex">
      <span class="text-success fs-5 fw-bold">15</span>
      <span class="mx-1 fs-5">Julio</span>
      <span class="mx-2 fw-light">2021, jueves</span>
    </div>

    <div class="entry-description">
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia,
      molestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum
      numquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium
      optio, eaque rerum! Provident similique accusantium nemo autem.
    </div>
  </div>
</template>

<script>
export default {
  name: 'DaybookEntry'
}
</script>

<style lang="scss" scoped>
.entry-container {
  border-bottom: 1px solid #2c3e50;
  transition: 0.2s all ease-in;

  &:hover {
    background-color: lighten($color: grey, $amount: 45);
    transition: 0.2s all ease-in;
  }

  .entry-description {
    font-size: 12px;
  }
}
</style>