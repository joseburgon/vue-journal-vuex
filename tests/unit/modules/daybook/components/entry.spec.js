import { shallowMount } from "@vue/test-utils";
import Entry from "@/modules/daybook/components/Entry";
import { journalState } from "../../../mock-data/test-journal-state";

describe("Entry component tests", () => {
  const mockRouter = {
    push: jest.fn(),
  };

  const wrapper = shallowMount(Entry, {
    props: {
      entry: journalState.entries[0],
    },
    global: {
      mocks: {
        $router: mockRouter,
      },
    },
  });

  test("should match the snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("should redirect when clicking the entry-container", () => {
    const entryContainer = wrapper.find('.entry-container')

    entryContainer.trigger('click')

    expect(mockRouter.push).toHaveBeenCalledWith(
      {
        name: 'entry',
        params: {
          id: journalState.entries[0].id
        }
      }
    )
  });

  test("computed properties tests", () => {
    expect(wrapper.vm.day).toBe(16)
    expect(wrapper.vm.month).toBe('June')
    expect(wrapper.vm.yearDay).toBe('2022, Thursday')
  });
});
