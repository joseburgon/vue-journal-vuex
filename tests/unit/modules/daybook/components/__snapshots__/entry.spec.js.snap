// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entry component tests should match the snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">16</span><span class="mx-1 fs-5">June</span></div>
  <div class="entry-description">Hello world from mock data</div>
</div>
`;
